agent/bench-scripts/pbench-run-benchmark.pl

```diff
@@ -36,19 +36,24 @@
 my $pp_only = 0;
 my $base_bench_dir = "";
 
+my @benchmarks = get_benchmark_names(get_pbench_bench_config_dir);
 # The only required [positional] argument is the benchmark name; verify it now
 if (scalar @ARGV == 0) {
-    print "You must supply at least a benchmark name:\n";
-    my @benchmarks = get_benchmark_names(get_pbench_bench_config_dir);
-    printf "%s\n",  join(" ", @benchmarks);
-    exit;
+    print "You must supply at least one of the following supported benchmark names:\n";
+    printf "\t%s\n",  join(" ", @benchmarks);
+    exit 1;
 }
 my $benchmark = shift(@ARGV);
 if ($benchmark eq "list") {
-    my @benchmarks = get_benchmark_names(get_pbench_bench_config_dir);
     printf "%s\n",  join(" ", @benchmarks);
     exit;
 }
+my %benchmarks_hash = map { $_ => 1 } @benchmarks;
```

Comment:
Ah, Perl idioms. I do wish I could have not recovered this arcane trivia...

Reply:
The sad thing is that I figured it out during the last review and now I cannot remember it any more...